portal/views/assessment_engine.py

```diff
@@ -1312,6 +1312,14 @@ def assessment_add(patient_id):
         if (qb and qn and (qn.id in [qbq.questionnaire.id
                            for qbq in qb.questionnaires])):
             qnr_qb = qb
+        # if a valid qb wasn't found, try the indefinite option
```

Comment:
My only worry with this is, what if the questionnaire is for the indefinite QB, but the same questionnaire type is in the user's `most_current_qb()`? Are we worried about the QNR getting assigned to the `most_current_qb()` instead of the indefinite? Or is this not a situation we're worried about (i.e. all the questionnaires for indefinite QBs, will never show up in non-indefinite QBs)?

Reply:
well, that situation would be tricky.  i'm comfortable waiting on that.

I would really like to see more aggressive validation here, but I'm afraid to break it at this hour.  Seems like we should not take submissions w/o a valid 'questionnaire' or a matching QB...